extension/encoding/awscloudwatchmetricstreamsencodingextension/opentelemetry_unmarshaler.go

```diff
@@ -4,15 +4,60 @@
 package awscloudwatchmetricstreamsencodingextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awscloudwatchmetricstreamsencodingextension"
 
 import (
+	"encoding/binary"
+	"errors"
 	"fmt"
 
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awscloudwatchmetricstreamsencodingextension/internal/metadata"
 )
 
-type formatOpenTelemetry10Unmarshaler struct{}
+var errInvalidUvarint = errors.New("invalid OTLP message length: failed to decode varint")
+
+type formatOpenTelemetry10Unmarshaler struct {
+	buildInfo component.BuildInfo
+}
 
 var _ pmetric.Unmarshaler = (*formatOpenTelemetry10Unmarshaler)(nil)
 
-func (f formatOpenTelemetry10Unmarshaler) UnmarshalMetrics(_ []byte) (pmetric.Metrics, error) {
-	return pmetric.Metrics{}, fmt.Errorf("UnmarshalMetrics unimplemented for format %q", formatOpenTelemetry10)
+func (f *formatOpenTelemetry10Unmarshaler) UnmarshalMetrics(record []byte) (pmetric.Metrics, error) {
+	md := pmetric.NewMetrics()
+	dataLen, pos := len(record), 0
+	for pos < dataLen {
+		// get size of datum
+		nLen, bytesRead := binary.Uvarint(record[pos:])
+		if bytesRead <= 0 {
+			return md, errInvalidUvarint
+		}
+		pos += bytesRead
+
+		// unmarshal datum
+		req := pmetricotlp.NewExportRequest()
+		err := req.UnmarshalProto(record[pos : pos+int(nLen)])
+		pos += int(nLen)
+		if err != nil {
+			return pmetric.Metrics{}, fmt.Errorf("unable to unmarshal input: %w", err)
+		}
```

Comment:
```suggestion
		if err := req.UnmarshalProto(record[pos : pos+int(nLen)]); err != nil {
			return pmetric.Metrics{}, fmt.Errorf("unable to unmarshal input: %w", err)
		}
		pos += int(nLen)
```

nit: check err first, keep err scoped to the `if` statement if possible